cookbooks/aws-parallelcluster-slurm/files/default/config_slurm/scripts/health_checks/gpu_health_check.sh

```diff
@@ -0,0 +1,209 @@
+    if [[ ! ${persistence_mode} == *"Enabled"* ]]; then
+      sudo nvidia-smi -i $gpu_id -pm 1
+      declare persistence_mode_enabled_$gpu_id=true
+    fi
+
+    ## Log useful GPU information
+    log_info "GPU : $gpu_id"
+    log_info "$(echo "$nvidia_smi_out" | grep "GPU UUID")"
+    log_info "$(echo "$nvidia_smi_out" | grep "Serial Number")"
+  done
+
+  ## Run GPUs Health Check test
+  log_info "Running GPU Health Check tool"
+  dcgmi diag -i $GPU_DEVICE_ORDINAL -r 2
+  dcgmi_exit_code=$(echo $?)
+
+  if [ $dcgmi_exit_code -ne 0 ]; then
+    if [ $dcgmi_exit_code -eq 127 ]; then
+      log_info "Health check has been executed but the DCGM Diagnostic tool is not available in this system or is located in a different file path. If possible use an official Parallel Cluster AMI."
+      dcgmi_exit_code=0
+    fi
+  else
+    log_info "The GPU Health Check succeeded"
+  fi
+  tear_down
+  return $dcgmi_exit_code
+}
+
+# FIXME REMOVE
+function install_dcgmi() {
+  # https://docs.nvidia.com/datacenter/dcgm/latest/user-guide/getting-started.html
+  # https://developer.download.nvidia.com/compute/cuda/repos/
+  # Alinux2
+  cd /etc/yum.repos.d/
+  sudo wget https://developer.download.nvidia.com/compute/cuda/repos/rhel7/x86_64/cuda-rhel7.repo
+  sudo yum install -y datacenter-gpu-manager
+}
+# END FIXME REMOVE
+
+function _log() {
+  echo "$(date +"%Y-%m-%d %H-%M-%S,%3N") - [${0##*/}] - $1 - JobID $SLURM_JOB_ID - " "${@:2}"
+}
+
+function log_info() {
+  _log "INFO" "$@"
+}
+
+function log_error() {
+  _log "ERROR" "$@"
+}
+
+function fast_success_exit() {
+  if [ $1 -ne 0 ]; then
+    log_info "$2"
+    tear_down
+    exit 0
+  fi
+}
+
+function is_nvidia()
+{
+  # Instances from https://aws.amazon.com/ec2/instance-types/
+  case "${instance_type}" in
+    g2.*|g3.*|g3s.*|g4dn.*|g5.*|g5g.*|p2.*|p3.*|p3dn.*|p4d.*|p4de.*) return ;;
+    *) return 1 ;;  # Not NVIDIA
+  esac
+}
+
+function set_instance_type()
+{
+  readonly token_url="http://169.254.169.254/latest/api/token"
+  readonly token_ttl_header="X-aws-ec2-metadata-token-ttl-seconds: 21600"
+  readonly instance_type_url="http://169.254.169.254/latest/meta-data/instance-type"
+  readonly common_opts=( --fail --silent --show-error )
+  local token
+  if ! token=$(curl "${common_opts[@]}" --request PUT --header "${token_ttl_header}" "${token_url}"); then
+    log_info "Failed to obtain IMDSv2 API token to determine instance type from metadata"
+  fi
+  readonly token_header="X-aws-ec2-metadata-token: ${token}"
+  if ! instance_type=$(curl "${common_opts[@]}" --header "${token_header}" "${instance_type_url}"); then
+    log_info "Failed to determine instance type from metadata"
+    return 1
+  fi
+}
+
+function start_service() {
+  sudo systemctl start $1
+  if [ $? -ne 0 ]; then
+    log_info "Unable to start service $1"
+    return 1
+  fi
+}
+
+function initialize_dcgm() {
+  log_info "Starting nvidia-dcgm service if not already started"
```

Comment:
Instead of logging this, what about logging when
* nvidia-dcgm is already started and we do nothing
* we start nvidia-dcgm because isn't already started

This way we better track down the execution steps